cmd/soroban-rpc/internal/db/event.go

```diff
@@ -97,3 +164,185 @@ func (eventHandler *eventHandler) InsertEvents(lcm xdr.LedgerCloseMeta) error {
+		}
+		if len(orConditions) > 0 {
+			rowQ = rowQ.Where(orConditions)
+		}
+	}
+
+	rows, err := eventHandler.db.Query(ctx, rowQ)
+	if err != nil {
+		return fmt.Errorf(
+			"db read failed for start ledger cursor= %v end ledger cursor= %v "+
+				"contractIDs= %v eventTypes= %v topics= %v   : %w",
+			cursorRange.Start.String(),
+			cursorRange.End.String(),
+			contractIDs,
+			eventTypes,
+			topics,
+			err)
+	}
+
+	defer rows.Close()
+
+	foundRows := false
+	for rows.Next() {
+		foundRows = true
+		var row struct {
+			eventCursorID   string `db:"id"`
+			eventData       []byte `db:"event_data"`
+			transactionHash []byte `db:"transaction_hash"`
+			ledgerCloseTime int64  `db:"ledger_close_time"`
+		}
+
+		err = rows.Scan(&row.eventCursorID, &row.eventData, &row.transactionHash, &row.ledgerCloseTime)
+		if err != nil {
+			return fmt.Errorf("failed to scan row: %w", err)
+		}
+
+		id, eventData, ledgerCloseTime := row.eventCursorID, row.eventData, row.ledgerCloseTime
+		transactionHash := row.transactionHash
+		cur, err := ParseCursor(id)
+		if err != nil {
+			return fmt.Errorf("failed to parse cursor: %w", err)
+		}
+
+		var eventXDR xdr.DiagnosticEvent
+		err = xdr.SafeUnmarshal(eventData, &eventXDR)
+		if err != nil {
+			return fmt.Errorf("failed to decode event: %w", err)
+		}
+		txHash := xdr.Hash(transactionHash)
+		if !f(eventXDR, cur, ledgerCloseTime, &txHash) {
+			return nil
+		}
+	}
+	if !foundRows {
+		eventHandler.log.
+			WithField("duration", time.Since(start)).
+			WithField("start", cursorRange.Start.String()).
+			WithField("end", cursorRange.End.String()).
+			WithField("contracts", contractIDs).
+			WithField("eventTypes", eventTypes).
+			WithField("Topics", topics).
+			Debugf(
+				"No events found for ledger range: duration= %v start ledger cursor= %v - end ledger cursor= %v contractIDs= %v",
+				time.Since(start),
+				cursorRange.Start.String(),
+				cursorRange.End.String(),
+				contractIDs,
+			)
+	}
+
+	eventHandler.log.
+		WithField("startLedgerSequence", cursorRange.Start.Ledger).
+		WithField("endLedgerSequence", cursorRange.End.Ledger).
+		WithField("duration", time.Since(start)).
+		Debugf("Fetched and decoded all the events with filters - contractIDs: %v ", contractIDs)
+
+	return nil
+}
+
+type eventTableMigration struct {
+	firstLedger uint32
+	lastLedger  uint32
+	writer      EventWriter
+}
+
+func (e *eventTableMigration) ApplicableRange() *LedgerSeqRange {
+	return &LedgerSeqRange{
+		FirstLedgerSeq: e.firstLedger,
+		LastLedgerSeq:  e.lastLedger,
+	}
+}
+
+func (e *eventTableMigration) Apply(_ context.Context, meta xdr.LedgerCloseMeta) error {
+	return e.writer.InsertEvents(meta)
+}
+
+func newEventTableMigration(
+	logger *log.Entry,
+	retentionWindow uint32,
+	passphrase string,
+) migrationApplierFactory {
+	return migrationApplierFactoryF(func(db *DB, latestLedger uint32) (MigrationApplier, error) {
+		firstLedgerToMigrate := firstLedger
```

Comment:
Am I blind or where is `firstLedger` defined?

Reply:
Migration code has been merged so this should be all good now.